From 4dbc5363d8036af09f5d361001c3d9d3799dfca7 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Tue, 27 Dec 2011 19:17:51 +0100
Subject: [PATCH 49/63] carl9170 firmware: minor wlan cleanup

 * replace hdr->seq_ctrl voodoo with use ieee80211_is_first_frag

 * add comment about consume_retry

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 carlfw/src/wlan.c |    8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

--- a/carlfw/src/wlan.c
+++ b/carlfw/src/wlan.c
@@ -305,7 +305,7 @@ static void wlan_assign_seq(struct ieee8
 	hdr->seq_ctrl &= cpu_to_le16(~IEEE80211_SCTL_SEQ);
 	hdr->seq_ctrl |= cpu_to_le16(fw.wlan.sequence[vif]);
 
-	if (!(hdr->seq_ctrl & cpu_to_le16(IEEE80211_SCTL_FRAG)))
+	if (ieee80211_is_first_frag(hdr->seq_ctrl))
 		fw.wlan.sequence[vif] += 0x10;
 }
 
@@ -351,6 +351,12 @@ static bool wlan_tx_status(struct dma_qu
 		/* reset retry indicator flags */
 		desc->ctrl &= ~(AR9170_CTRL_TXFAIL | AR9170_CTRL_BAFAIL);
 
+		/*
+		 * Note: wlan_tx_consume_retry will override the old
+		 * phy [CCK,OFDM, HT, BW20/40, MCS...] and mac vectors
+		 * [AMPDU,RTS/CTS,...] therefore be careful when they
+		 * are used.
+		 */
 		if (wlan_tx_consume_retry(super)) {
 			/*
 			 * retry for simple and aggregated 802.11 frames.
